subsys/bluetooth/mesh/provisioner.c

```diff
@@ -698,10 +700,7 @@ int bt_mesh_auth_method_set_static(const uint8_t *static_val, uint8_t size)
 	prov_set_method(AUTH_METHOD_STATIC, 0, 0);
 
 	memcpy(bt_mesh_prov_link.auth + 16 - size, static_val, size);
-	if (size < 16) {
-		(void)memset(bt_mesh_prov_link.auth, 0,
-			     sizeof(bt_mesh_prov_link.auth) - size);
-	}
```

Comment:
Why is this part being removed? I don't see an explanation of it in the commit message (or then I didn't understand that part in the commit message).

Reply:
When every call `bt_mesh_auth_method_set<*>`
should also clear auth value, otherwise will case confirm
failed.

Reply:
@LingaoM sorry, I'm still not understanding. First of all, I'm not seeing any clearing in the other set() methods like you say. Secondly, all this memset was doing was to ensure that no garbage is left in `bt_mesh_prov_link.auth` if the new size is smaller than the previous one.

Reply:
This function, or indeed others in the `bt_mesh_auth_method_set_*` do **not** clear the `auth` array.